src/main/java/org/folio/rest/impl/PutOrdersByIdHelper.java

```diff
@@ -4,109 +4,151 @@
+                AsyncResult<javax.ws.rs.core.Response> result = Future.succeededFuture(response);
+                asyncResultHandler.handle(result);
+              })
+              .exceptionally(this::handleError);
+          })
+          .exceptionally(this::handleError);
+      })
+      .exceptionally(this::handleError);
+
     return future;
   }
 
-  public Void handleError(Throwable throwable) {
-    final Future<javax.ws.rs.core.Response> result;
-
-    logger.error("Exception while updating orders", throwable.getCause());
+  private CompletableFuture<Void> handlePoLines(CompositePurchaseOrder compOrder, JsonArray poLinesFromStorage) {
+    List<CompletableFuture<?>> futures = new ArrayList<>();
+    List<PoLine> linesToCreate = new ArrayList<>();
+    if (poLinesFromStorage.isEmpty()) {
+      linesToCreate.addAll(compOrder.getPoLines());
+    } else {
+      futures.addAll(filterPoLinesAndUpdateMatching(compOrder, poLinesFromStorage, linesToCreate));
+    }
+    poLinesFromStorage.stream().forEach(poLine ->futures.add(deletePoLine((JsonObject) poLine, httpClient, ctx, okapiHeaders, logger)));
+    boolean updateInventory = compOrder.getWorkflowStatus() == CompositePurchaseOrder.WorkflowStatus.OPEN;
+    linesToCreate.forEach(poLine -> futures.add(new PostOrderLineHelper(httpClient, okapiHeaders, asyncResultHandler, ctx)
+      .createPoLine(poLine, updateInventory)));
+    return VertxCompletableFuture.allOf(ctx, futures.toArray(new CompletableFuture[0]));
+  }
 
-    final Throwable t = throwable.getCause();
-    if (t instanceof HttpException) {
-      final int code = ((HttpException) t).getCode();
-      final String message = ((HttpException) t).getMessage();
-      switch (code) {
-        case 404:
-          result = Future.succeededFuture(PutOrdersByIdResponse.respond404WithTextPlain(message));
-          break;
-        case 500:
-          result = Future.succeededFuture(PutOrdersByIdResponse.respond500WithTextPlain(message));
+  private List<CompletableFuture<?>> filterPoLinesAndUpdateMatching(CompositePurchaseOrder compOrder, JsonArray poLinesFromStorage, List<PoLine> linesToCreate) {
+    List<CompletableFuture<?>> futures = new ArrayList<>();
+    for (PoLine line : compOrder.getPoLines()) {
+      JsonObject existingLine = null;
+      for (int i = 0; i < poLinesFromStorage.size(); i++) {
+        JsonObject lineFromStorage = poLinesFromStorage.getJsonObject(i);
+        if (StringUtils.equals(lineFromStorage.getString("id"), line.getId())) {
+          existingLine = lineFromStorage;
+          poLinesFromStorage.remove(i);
           break;
-        default:
-          result = Future.succeededFuture(PutOrdersByIdResponse.respond500WithTextPlain(message));
+        }
+      }
+      if (existingLine == null ) {
+        linesToCreate.add(line);
+      } else {
+        futures.add(putLineHelper.updateOrderLine(line, JsonObject.mapFrom(existingLine)));
```

Comment:
As far as I can see `existingLine` is `JsonObject`. Is `JsonObject.mapFrom` still required?